projects/rocprofiler-systems/source/lib/core/trace_cache/perfetto_processor.cpp

```diff
@@ -0,0 +1,1142 @@
+        }
+        return result;
+    };
+
+    static std::once_flag init_flag;
+    std::call_once(init_flag, []() {
+        process_page_track::emplace(0, "CPU Memory Usage (S)", "MB");
+        process_virt_track::emplace(0, "CPU Virtual Memory (S)", "MB");
+        process_peak_track::emplace(0, "CPU Peak Memory (S)", "MB");
+        process_cntx_track::emplace(0, "CPU Context Switches (S)", "");
+        process_flts_track::emplace(0, "CPU Page Faults (S)", "");
+        process_user_track::emplace(0, "CPU User Time (S)", "sec");
+        process_kern_track::emplace(0, "CPU Kernel Time (S)", "sec");
+    });
+
+    auto _ts = _cpu_sample.timestamp;
+
+    TRACE_COUNTER(trait::name<category::process_page>::value,
+                  process_page_track::at(0, 0), _ts,
+                  static_cast<double>(_cpu_sample.page_rss) / units::megabyte);
+
+    TRACE_COUNTER(trait::name<category::process_virt>::value,
+                  process_virt_track::at(0, 0), _ts,
+                  static_cast<double>(_cpu_sample.virt_mem_usage) / units::megabyte);
+
+    TRACE_COUNTER(trait::name<category::process_peak>::value,
+                  process_peak_track::at(0, 0), _ts,
+                  static_cast<double>(_cpu_sample.peak_rss) / units::megabyte);
+
+    TRACE_COUNTER(trait::name<category::process_context_switch>::value,
+                  process_cntx_track::at(0, 0), _ts,
+                  static_cast<double>(_cpu_sample.context_switch_count));
+
+    TRACE_COUNTER(trait::name<category::process_page_fault>::value,
+                  process_flts_track::at(0, 0), _ts,
+                  static_cast<double>(_cpu_sample.page_faults));
+
+    TRACE_COUNTER(trait::name<category::process_user_mode_time>::value,
+                  process_user_track::at(0, 0), _ts,
+                  static_cast<double>(_cpu_sample.user_mode_time) / units::sec);
+
+    TRACE_COUNTER(trait::name<category::process_kernel_mode_time>::value,
+                  process_kern_track::at(0, 0), _ts,
+                  static_cast<double>(_cpu_sample.kernel_mode_time) / units::sec);
+
+    auto cpu_freqs = deserialize_freqs(_cpu_sample.freqs);
+    for(const auto& cpu_data : cpu_freqs)
+    {
+        size_t cpu_id = cpu_data.id;
+        if(!cpu_freq_track::exists(cpu_id))
+        {
+            std::string track_name = "CPU Frequency [" + std::to_string(cpu_id) + "] (S)";
+            cpu_freq_track::emplace(cpu_id, track_name, "MHz");
+        }
+
+        TRACE_COUNTER(trait::name<category::cpu_freq>::value,
+                      cpu_freq_track::at(cpu_id, 0), _ts,
+                      static_cast<double>(cpu_data.value));
+    }
+}
+
+void
+perfetto_processor_t::handle([[maybe_unused]] const backtrace_region_sample& _bts)
+{
+    (_bts.category == trait::name<category::timer_sampling>::value)
+        ? write_track_data<category::timer_sampling>(_bts, m_use_annotations)
+        : write_track_data<category::overflow_sampling>(_bts, m_use_annotations);
```

Comment:
```suggestion
    constexpr auto _timer_category { trait::name<category::timer_sampling>::value };
    
    if(_bts.category == _timer_category) {
         write_track_data<category::timer_sampling>(_bts, m_use_annotations)
    }
    else {
         write_track_data<category::overflow_sampling>(_bts, m_use_annotations);
    }
```

I guess ternary operator instead of `if statemens` is not style in the project.
And it's also harder to read.

Reply:
Personally, I prefer ternary operator in cases as such, simply because it keeps the code cleaner. If you really think that it's nor readable or inline with the current project style, I don't mind replacing it with `if-else` block